libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -252,15 +249,37 @@ func setICMP(ipv iptables.IPVersion, bridgeName string, enable bool) error {
 	return appendOrDelChainRule(icmpRule, "ICMP", enable)
 }
 
-// Append to the filter table's DOCKER chain (the default DROP rule must follow
-// per-port ACCEPT rules, which will be inserted at the top of the chain).
-func defaultDrop(ipv iptables.IPVersion, bridgeName string, enable bool) error {
-	dropRule := iptRule{ipv: ipv, table: iptables.Filter, chain: DockerChain, args: []string{
+func (n *bridgeNetwork) setDefaultForwardRule(
+	ipVersion iptables.IPVersion,
+	bridgeName string,
+) error {
+	// Normally, DROP anything that hasn't been ACCEPTed by a per-port/protocol
+	// rule. This prevents direct access to un-mapped ports from remote hosts
+	// that can route directly to the container's address (by setting up a
+	// route via the host's address).
+	action := "DROP"
+	if n.isGwModeUnprotected(ipVersion) {
+		// If the user really wants to allow all access from the wider network,
+		// explicitly ACCEPT anything so that the filter-FORWARD chain's
+		// default policy can't interfere.
+		action = "ACCEPT"
+	}
+
+	rule := iptRule{ipv: ipVersion, table: iptables.Filter, chain: DockerChain, args: []string{
 		"!", "-i", bridgeName,
 		"-o", bridgeName,
-		"-j", "DROP",
+		"-j", action,
 	}}
-	return appendOrDelChainRule(dropRule, "DEFAULT DROP", enable)
+
+	// Append to the filter table's DOCKER chain (the default rule must follow
+	// per-port ACCEPT rules, which will be inserted at the top of the chain).
+	if err := appendOrDelChainRule(rule, "DEFAULT FWD", true); err != nil {
+		return fmt.Errorf("failed to add default-drop rule: %s", err.Error())
```

Comment:
ugh; I see a bunch more further up, and this was probably existing code, but we really shouldn't use `err.Error()` for these, and either wrap the error (if we need to be able to access the wrapped one) or just format;

```suggestion
		return fmt.Errorf("failed to add default-drop rule: %w", err)
```

or (if we want to mask the underlying error)

```suggestion
		return fmt.Errorf("failed to add default-drop rule: %v", err)
```

Reply:
I guess it was a copy and paste of an existing block - I'll change it.